### ISSUE TITLE
Packages from environment.yml not installed into python3 env

### ISSUE THREAD

**Issue by astrojuanlu** [binder-project/binder#54]
I have a directory with Python 3 notebooks, and I cannot directly run them even though the dependencies are listed in the corresponding `environment.yml`. This is not a big problem, since I can just instruct users to install them again, but it's a bit annoying.

This is the repository:

https://github.com/Juanlu001/poliastro/blob/master/environment.yml

And these are the available packages in each environment:

https://gist.github.com/Juanlu001/0abb2f5a09d8b549da08

**Comment by maxalbert** [comment]
I'm having the same problem. This is my repository:

https://github.com/maxalbert/paper-supplement-nanoparticle-sensing

The corresponding binder is here:

http://mybinder.org/repo/maxalbert/paper-supplement-nanoparticle-sensing

If I am not mistaken then the problem seems to be that the conda environment created by binder is somehow not activated in the docker container (even though the log file claims that it is). For example, when I open one of my notebooks and run `!conda env list` in a new cell, this is the output I get:

```
# conda environments:
#
binder                   /home/main/anaconda2/envs/binder
python3                  /home/main/anaconda2/envs/python3
root                  *  /home/main/anaconda2
```

The `binder` environment is the one created from my `environment.yml` file, but the root environment is the one that is activated. Any chance to fix this? Many thanks!

**Comment by andrewosh** [comment]
@Juanlu001 @maxalbert thanks for opening up this issue. The `environment.yml` handling that we're doing right now has definitely been a source of confusion, so we really should add some documentation here. 

The immediate fix to your problem is to explicitly specify the Python version you'd like to use inside of your `environment.yml` file (just like here: https://github.com/binder-project/example-conda-environment/blob/master/environment.yml). With that line included, the default kernel will be Python 3.4, and the packages in the file will be available.

If you check out how we handle the files here: https://github.com/binder-project/binder/blob/master/binder/app.py#L204, you'll see that we aren't explicitly `source activate`ing the environment, because that doesn't behave well during the Docker build process, but we are manipulating the path in such a way that the `binder` environment the first environment searched for packages.

From inside the `binder-project/example-conda-environment` binder:

```
import sys
sys.path
```

```
['',
 '/home/main/anaconda2/envs/binder/lib/python34.zip',
 '/home/main/anaconda2/envs/binder/lib/python3.4',
 '/home/main/anaconda2/envs/binder/lib/python3.4/plat-linux',
 '/home/main/anaconda2/envs/binder/lib/python3.4/lib-dynload',
 '/home/main/anaconda2/envs/binder/lib/python3.4/site-packages/setuptools-20.7.0-py3.4.egg',
 '/home/main/anaconda2/envs/binder/lib/python3.4/site-packages',
 '/home/main/anaconda2/envs/binder/lib/python3.4/site-packages/IPython/extensions',
 '/home/main/.ipython']
```

**Comment by astrojuanlu** [comment]
Thank you @andrewosh, adding `python=3.4` to `environment.yml` indeed fixed the issue. Also your explanation on the file handling was spot on, since `!conda list` won't show the packages available from the path and that can be confusing. Closing!
